tracing-subscriber/src/registry/sharded.rs

```diff
@@ -0,0 +1,280 @@
+use sharded_slab::{Guard, Slab};
+
+use crate::{
+    fmt::span::SpanStack,
+    registry::{LookupSpan, SpanData},
+};
+use serde_json::{json, Value};
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    convert::TryInto,
+    fmt,
+    sync::{
+        atomic::{fence, AtomicUsize, Ordering},
+        Arc, Mutex,
+    },
+};
+use tracing_core::{
+    dispatcher,
+    field::{FieldSet, Visit},
+    span::{self, Id},
+    Event, Field, Interest, Metadata, Subscriber,
+};
+
+#[derive(Debug)]
+pub struct Registry {
+    spans: Arc<Slab<Data>>,
+}
+
+thread_local! {
+    static CONTEXT: RefCell<SpanStack> = RefCell::new(SpanStack::new());
+}
+
+#[derive(Debug)]
+pub struct Data {
+    metadata: &'static Metadata<'static>,
+    parent: Option<Id>,
+    children: Vec<Id>,
```

Comment:
this will probably need a lock or something? although I'm on the fence about whether we even want to have children in the registry API.

Reply:
I think it will, and I don't think `clone_span` is sufficient.

Reply:
this makes the lifetimes complex at the moment, but I'll tackle this later, as it's not used at the moment.